src/fcntl.rs

```diff
@@ -797,6 +797,9 @@ pub enum FcntlArg<'a> {
     /// Return the full path without firmlinks of the fd.
     #[cfg(apple_targets)]
     F_GETPATH_NOFIRMLINK(&'a mut PathBuf),
+    /// Issue an advisory read async with no copy to user
+    #[cfg(apple_targets)]
+    F_RDADVISE(libc::radvisory),
```

Comment:
I saw this was changed in the "testing" commit, the previous API is indeed a bit unclear:

```
F_RDADVISE(off_t, c_int),
```

I am actually thinking about this:

```rs
F_RDADVISE {
    offset: off_t,
    count: c_int,
}
```

So semantically, the API will be more clear and users won't need to bother with the libc raw structure. The drawback is, we need to copy these 2 numbers, considering they are simply numbers, IMHO, it is fine to copy them. Your thoughts? 🤔

Reply:
I was being consistent with flags passing `libc::flock`.

Reply:
Get it, makes sense to me then:)